page_iterator.go

```diff
@@ -0,0 +1,77 @@
+package msgraphgocore
+
+import "net/http"
+
+type Item interface {
+	GetDisplayName() string
+}
+
+type Page interface {
+	GetValue() []Item
+	GetNextLink() string
+	GetNextPage() Page
+}
+
+type PageIterator struct {
+	page       Page
+	client     http.Client
+	pauseIndex int
+}
+
+func NewPageIterator(page Page, client http.Client) *PageIterator {
```

Comment:
should accept a [GraphRequestAdapterBase](https://github.com/microsoftgraph/msgraph-sdk-go-core/blob/main/graph_request_adapter_base.go) instead so you can use the existing infrastructure (request information and the passed request adapter) to handle making the http call and deserializing the result for you.